test/units/module_utils/test_api.py

```diff
@@ -46,3 +46,11 @@ def login_database():
 
         with pytest.raises(Exception):
             login_database()
+
+    def test_no_retries(self):
+
+        @retry()
+        def login_database():
+            raise Exception("Should not execute")
```

Comment:
I believe this could be
```suggestion
            assert False, "Should not execute"
```

Reply:
Or
```suggestion
            pytest.fail("Should not execute")
```